site/docs/v1/tech/oauth/integrations/salesforce.adoc

```diff
@@ -0,0 +1,213 @@
+        u.profileId = p.Id;
+        u.username = data.email + ORG_SUFFIX;
+        u.email = data.email;
+        if (data.firstName != null && data.firstName != '') {
+            u.firstName = data.firstName;
+        } else {
+            String givenName = data.attributeMap.get('given_name');
+            if (givenName != null && givenName != '') {
+                u.firstName = givenName;
+            }
+        }
+        if (data.lastName != null && data.lastName != '') {
+            u.lastName = data.lastName;
+        } else {
+            String familyName = data.attributeMap.get('family_name');
+            if (familyName != null && familyName != '') {
+                u.lastName = familyName;
+            }
+        }
+        String alias;
+        if (data.userName != null && data.userName != '') {
+            alias = data.userName;
+        } else {
+            alias = data.email;
+        }
+        if (alias.length() > 8) {
+            alias = alias.substring(0, 8);
+        }
+        u.alias = alias;
+        u.languagelocalekey = UserInfo.getLocale();
+        u.localesidkey = UserInfo.getLocale();
+        u.emailEncodingKey = 'UTF-8';
+        u.timeZoneSidKey = data.attributeMap.get('timezone');
+        if (u.timeZoneSidKey == null || u.timeZoneSidKey == '') {
+            // @TODO change this to the default timezone for users
+            u.timeZoneSidKey = 'GMT';
+        }
+        return u;
+    }
+}
+```
+
+Change the `ORG_SUFFIX` in the sixth line from that code with your actual FusionAuth instance URL. You can also modify the `STANDARD_PROFILE` in the third line with the profile name you want to add users to (you can find the available options navigating to [breacrumb]#ADMINISTRATION -> Users -> Profiles#) and `u.timeZoneSidKey` value in the 60th line with a timezone link:https://help.salesforce.com/s/articleView?id=sf.admin_supported_timezone.htm&type=5[from this list] to be used as a fallback when not receiving one from FusionAuth. Finally, click [uielement]#Save# to create the class.
+
+Browse to [breadcrumb]#SETTINGS -> Identity -> Auth. Providers# to get to the Authentication Providers page and click [uielement]#New#.
+
+image::oauth/integrations/salesforce/auth-provider-new.png[Create an Authentication Provider.,width=1200,role=bottom-cropped]
+
+Fill in the values as shown in the image below, using the information you copied from your FusionAuth application earlier.
+
+image::oauth/integrations/salesforce/auth-provider-form.png[Fill in Authentication Provider values.,width=1200]
+
+The following table describes the necessary values that you need to copy from your FusionAuth application onto the fields in Salesforce.
+
+[cols="1,2,1"]
+|===
+| Salesforce field | FusionAuth field | Sample value
+
+| [field]#Provider Type#
+| `Open ID Connect`
+| {nbsp}
+
+| [field]#Name#
+| FusionAuth
+| {nbsp}
+
+| [field]#URL Suffix#
+| FusionAuth
+| {nbsp}
```

Comment:
Should the value include the leading `.`? Should this match the `ORG_SUFFIX` from the sample Apex code?

This change would require updating the screenshot if we want to keep them consistent.

```suggestion
| [field]#URL Suffix#
| {nbsp}
| `your.fusionauth.application.url`
```

Reply:
No, this is what would be added to the SSO URL (e.g. `https://your-application.my.salesforce.com/services/auth/sso/[URL-SUFFIX]`)